app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/upload/ElementEditsUploader.kt

```diff
@@ -50,9 +50,9 @@ class ElementEditsUploader @Inject constructor(
             mapDataController.updateAll(updates)
 
             if (edit.action is IsRevertAction) {
-                statisticsUpdater.subtractOne(questTypeName, edit.position)
+                statisticsUpdater.subtractOne(edit.questType, edit.position)
             } else {
-                statisticsUpdater.addOne(questTypeName, edit.position)
+                statisticsUpdater.addOne(edit.questType, edit.position)
```

Comment:
I changed the parameter type of `addOne` and `subtractOne` in `StatisticsUpdater` from `String` to `QuestType<*>` ([`c38b3b1` (#3327)](https://github.com/streetcomplete/StreetComplete/pull/3327/commits/c38b3b17b411915857e25e5e51549c04efd2a530#diff-365799f69fa1b3c05d09ec193821fcf4c7e86a0587c1e0c71131ac718d00eb8f)) because that calls `updateQuestTypeAchievements` in `AchievementGiver`, where I have done the same thing ([`c38b3b1` (#3327)](https://github.com/streetcomplete/StreetComplete/pull/3327/commits/c38b3b17b411915857e25e5e51549c04efd2a530#diff-f1ac26586b0ffe62a303960cdff7cc83f87520fcc1478449dd1cddbe41fbd1a0)).

Reply:
Now that the `AchievementsGiver` has access to the `QuestTypeRegistry` anyway, should I rather revert this change (i.e. keep passing the quest type name around instead of the quest type instance) and get the quest type instance via `questTypeRegistry.getByName()!!`?

Reply:
yeah